runner/internal/config/config.go

```diff
@@ -83,13 +84,34 @@ type Output struct {
 	Silent bool
 }
 
+type set map[string]struct{}
+
+func (set set) add(values ...string) {
+	for _, v := range values {
+		set[v] = struct{}{}
+	}
+}
+
 // Global represents the global configuration of the runner.
 // It must be validated using Global.Validate before usage.
 type Global struct {
 	Request Request
 	Runner  Runner
 	Output  Output
 	Tests   []tests.Case
+
+	fieldsSet set
```

Comment:
Suggestion: disambiguate "set" as the underlying data structure from the actual meaning "field is set by user/something, and is not at its default value".

```suggestion
	fieldsManuallySet set
```
or
```suggestion
	fieldsUndefaulted set
```